### ISSUE TITLE
[Feature] Remove required 'version' from dbt_project.yml

### ISSUE THREAD

**Issue by seub** [dbt-labs/dbt-core#6603]
### Is this your first time submitting a feature request?

- [X] I have read the [expectations for open source contributors](https://docs.getdbt.com/docs/contributing/oss-expectations)
- [X] I have searched the existing issues, and I could not find an existing issue for this feature
- [X] I am requesting a straightforward extension of existing dbt functionality, rather than a Big Idea better suited to a discussion

### Describe the feature

At the moment, it is required that dbt projects include a `version` in their dbt_project.yml, which should be a semantic version, as mentioned in the dbt doc [here](https://docs.getdbt.com/reference/project-configs/version).

My dbt project is a git package with its own process for creating and publishing its semantic versioning. The semantic version is not hard coded in the code base, but it is generated semi-automatically by our CI/CD. I imagine this is a common practice.

Therefore our dbt project has 2 semantic versions, the one in `dbt_project.yml`, which does not serve any meaningful purpose according to the dbt doc, and the git semantic version (tag).

In summary, I suggest removing the required 'version' from dbt_project.yml, because:
- it does not serve any purpose
- it is confusing as it is potentially in conflict with the git repo's own versioning.

### Who will this benefit?

All?

### Are you interested in contributing this feature?

Sure

### Anything else?

https://docs.getdbt.com/reference/project-configs/version

**Comment by dbeatty10** [comment]
Love this suggestion @seub !

I distinctly remember when a colleague enlightened me that `version` isn't used at all. `dbt_utils` is a great example of a dbt package that just [permanently set a dummy version number of `0.1.0`](https://github.com/dbt-labs/dbt-utils/blob/a6c3159af43e237a6a2057f43461d5040f1193ce/dbt_project.yml#L2) (even though its latest version is `1.0.0` as-of this writing).

Looking forward to hearing more of your thoughts @jtcohen6. (I still need to write-up our discussion we had about that _different_ mechanism -- will drop a link in this thread once I do so.)

Some things for us to consider:
- In our current world, would it be a breaking change if we relax `version` from required to optional? (I suspect "no")
- In a world with `dbt deps freeze`, how would we want to think about the "version" of dbt packages installed via git hash or git tag or tarball URL?

**Comment by jtcohen6** [comment]
@dbeatty10 Without stealing any thunder from the forthcoming write-up — given that the only good reason I could come up with ~a year ago for continuing to require `version` was, "maybe we'd use this for `deps`" — and given our chat about other better mechanisms for doing that (if we had capacity & appetite) — I struggle to think of any other good _future_ reason to keep requiring `version` today.

A colleague just asked a perfectly reasonable, related question:
> Why, when I configure `config-version: 2` in my dbt project file, do I still need to configure `version: 2` in my yml files?

These are entirely separate versioning schemes. The _reason_ we had to call it `config-version` in `dbt_project.yml`, back in v0.17 when we were making breaking changes to the `vars` structure... is because a field named `version` was already taken! Even though it served no purpose!

@seub So - I'd be open to relaxing this. This will just require a few targeted one-line updates, to make this `Optional` (`None` by default):

https://github.com/dbt-labs/dbt-core/blob/065ab2ebc2104635770f49ca48bc434532d3ef41/core/dbt/contracts/project.py#L187

https://github.com/dbt-labs/dbt-core/blob/065ab2ebc2104635770f49ca48bc434532d3ef41/core/dbt/config/project.py#L528

And we'd definitely want to update these docs: https://docs.getdbt.com/reference/project-configs/version#dbt_projectyml-versions

Would you be interested in contributing that change?

**Comment by seub** [comment]
@jtcohen6 Thank you! Yes, let me try to contribute that change.

**Comment by dbeatty10** [comment]
@seub Excited that you're going to work on that contribution! 🤩 

As promised, here's the write-up for that _different_ mechanism to unlock `dbt deps freeze` that @jtcohen6 and I discussed last week and was mentioned [here](https://github.com/dbt-labs/dbt-core/issues/6603#issuecomment-1382148495):
- https://github.com/dbt-labs/dbt-core/issues/6643

**Comment by seub** [comment]
@jtcohen6 @dbeatty10 Thank you for your encouragements! 

I'm stuck trying to get my dev setup to work for integration tests: `make setup-db` gives me errors `error: connection to server at "localhost" (::1), port 5432 failed: FATAL:  role "root" does not exist`. I asked for help on Slack [here](https://getdbt.slack.com/archives/C50NEBJGG/p1674001299554449), just mentioning it here for visibility in case either of you have an idea 🙏

**Comment by dbeatty10** [comment]
> I asked for help on Slack [here](https://getdbt.slack.com/archives/C50NEBJGG/p1674001299554449)

We'll see if we can get to the bottom of that error -- I just sent a reply in Slack.

**Comment by seub** [comment]
Hi all, it looks to me as though there is a bug in the current version of the Makefile in dbt-core, causing `make integration` to fail with errors like:
```bash
Env var required but not provided: 'DBT_TEST_USER_1'
```
Replacing `$(if $(USE_CI_FLAGS), $(CI_FLAGS))` by `$(CI_FLAGS)` on line 69 apparently solves the problem.

My friend @dsillman2000 who helped me figure this out just created an issue: https://github.com/dbt-labs/dbt-core/issues/6689

I guess I'll wait for the main branch to fix this bug and fork the repo again for this issue.

**Comment by seub** [comment]
Hi all, I created a [PR](https://github.com/dbt-labs/dbt-core/pull/6711) to make the change. Thanks for your help @jtcohen6 and @dbeatty10 !

I did not add any tests. The current tests all passed. If you'd like me to add a unit or integration test, a tip would be great.

I also created an [issue to update the doc](https://github.com/dbt-labs/docs.getdbt.com/issues/2747).

**Comment by seub** [comment]
My PR was broken due to an email mistake, I just replaced it with a new PR: https://github.com/dbt-labs/dbt-core/pull/6718
